AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Internal/URLComponents+sigv4Encoding.swift

```diff
@@ -22,15 +22,29 @@ extension URLComponents {
         var sigV4UnreservedCharacters = CharacterSet(charactersIn: "A" ... "Z")
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "a" ... "z"))
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "0" ... "9"))
-        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~.%=&"))
+        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~."))
         return sigV4UnreservedCharacters
     }()
 
-    mutating func percentEncodeQueryBySigV4Rules() {
-        guard let percentEncodedQuery = self.percentEncodedQuery else {
+    private func encodeQueryParamValueBySigV4Rules(_ value: String) -> String {
+        /// removingPercentEncoding returns `nil` if called on a value
+        /// that hasn't been prior encoded
+        let unencoded = value.removingPercentEncoding ?? value
+
+        /// apply sigv4 encoding
+        return unencoded.addingPercentEncoding(
+            withAllowedCharacters: Self.sigV4UnreservedCharacters) ?? value
+    }
+
+    mutating func queryItemsEncodedPerSigV4Rules(_ queryItems: [String: String]?) {
+        guard let queryItems = queryItems else {
             return
         }
-        self.percentEncodedQuery = percentEncodedQuery.addingPercentEncoding(
-            withAllowedCharacters: Self.sigV4UnreservedCharacters)
+        percentEncodedQuery = queryItems.map { (name, value) -> String in
```

Comment:
Style nit: is the full type signature required here? (It might be--Swift's type inference is great all the way up until it isn't. :) )

```suggestion
        percentEncodedQuery = queryItems.map { name, value in
```